hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceShipper.java

```diff
@@ -122,6 +122,10 @@ public final void run() {
       } catch (InterruptedException e) {
         LOG.trace("Interrupted while waiting for next replication entry batch", e);
         Thread.currentThread().interrupt();
+      } catch (ReplicationRuntimeException rre) {
```

Comment:
I think we could do this
```
catch (InterruptedException | ReplicationRuntimeException e) {
  blabla
}
```

Reply:
Will do, and updated the patch to fix the NullPointerException as well, thanks.